File: mobile/lib/features/admin/locations/data/models/district.dart
/// Mirrors `DistrictResource`.
class District {
  District({
    required this.id,
    required this.provinceId,
    required this.name,
    required this.nameNe,
    required this.code,
  });

  factory District.fromJson(Map<String, dynamic> json) {
    return District(
      id: json['id'] as int,
      provinceId: json['province_id'] as int,
      name: json['name'] as String,
      nameNe: json['name_ne'] as String?,
      code: json['code'] as String,
    );
  }

  final int id;
  final int provinceId;
  final String name;
  final String? nameNe;
  final String code;
}

File: backend/app/Models/District.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class District extends Model
{
    protected $fillable = ['province_id', 'name', 'name_ne', 'code'];

    protected function casts(): array
    {
        return [
            'province_id' => 'integer',
        ];
    }

    public function province(): BelongsTo
    {
        return $this->belongsTo(Province::class);
    }

    public function municipalities(): HasMany
    {
        return $this->hasMany(Municipality::class);
    }
}
